.github/workflows/go.yaml

```diff
@@ -50,7 +50,31 @@ jobs:
     strategy:
       matrix: ${{fromJson(needs.prepare.outputs.matrix)}}
       fail-fast: false
+
+    # env:
+    #   DOCKER_VERSION: 24.0.9
+
     steps:
+    - name: Install Docker ${{ env.DOCKER_VERSION }}
+      if: env.DOCKER_VERSION != ''
```

Comment:
Maybe this needs some kind of condition on the runs-on?

Reply:
Not sure what you mean. What do you have in mind?

Reply:
This only works on debian-derivatives.

But maybe it doesn't matter. If you change the runs-on, you fix the rest of the workflow to work with the new value.